src/io/result_io.jl

```diff
@@ -262,9 +262,16 @@ function setup_result_store!(domain::Domain, scen_spec::DataFrame)::Tuple
     inputs = zcreate(Float64, input_dims...; fill_value=-9999.0, fill_as_missing=false, path=input_loc, chunks=input_dims, attrs=attrs)
 
     # Store post-processed table of input parameters.
-    # +1 skips the RCP column
-    integer_params = findall(_check_discrete.(domain.model[:ptype]))
-    map_to_discrete!(scen_spec[:, integer_params.+1], getindex.(domain.model[:bounds], 2)[integer_params])
+    ms = model_spec(domain)
+
+    # Find discrete parameters that are not constant
+    discrete_param_idx = findall(_check_discrete.(ms.ptype) .& .!ms.is_constant)
+    discrete_params = @view scen_spec[!, Not(:RCP)][!, discrete_param_idx]
+
+    # Map continuous values to discrete, and fill scenario spec with updated values.
```

Comment:
This comment may not be that useful 🤷

Reply:
Yeah I guess the function name explains what it does already